File: dataset/human/python/file_8895.py
import Control.Applicative ((<$>), (<*>))
import Control.Monad
import Data.List

main :: IO ()
main = do
  solve <$> getLine >>= putStrLn

solve :: String -> String
solve xs = f $ reverse xs
  where
    f "" = "YES"
    f ss | length ss >= 5 && take 5 ss == "maerd" = f (drop 5 ss)
         | length ss >= 5 && take 5 ss == "esare" = f (drop 5 ss)
         | length ss >= 6 && take 6 ss == "resare" = f (drop 6 ss)
         | length ss >= 7 && take 7 ss == "remaerd" = f (drop 7 ss)
         | otherwise = "NO"

File: dataset/human/c/file_8895.c
#include <stdio.h>

int main()
{
	int h, w;
	int i, j;
	char p[52][53]={NULL};

	scanf("%d%d", &h, &w);

	for(i=0; i < h; i++){
		scanf("%s", &p[i+1][1]);
	}

	for(i=1; i <= h; i++){
		for(j=1; j <= w; j++){
			if(p[i][j] == '#' && p[i-1][j] != '#' && p[i+1][j] != '#' && p[i][j-1] != '#' && p[i][j+1] != '#'){
				break;
			}
		}
		if(j != w+1){break;}
	}


	if(i == h+1 && j == w+1){
		printf("Yes");
	}
	else{
		printf("No");
	}


return 0;
}
